### PR TITLE
Blog App : Milestone : N+1 problems integration tests

### PR DESCRIPTION
**Pull request by ahmed025566**
Hello there, Code Reviewer
Hope you are doing well.

In this milestone you could see that I have done the following: 
- I have optimized performance by decreasing the number of quires : I have fetched all the data I want in the controller and then I have used them in the view by loops and if statements and here is the quires before and after: 
- Before
before optmization
show before optmization
- After 

show after optmization
after optmization

- I have written all the required test cases for user and post both show and index pages.

### CODE REVIEW

File: app/controllers/posts_controller.rb [hunk 1]
```diff
@@ -2,11 +2,14 @@ class PostsController < ApplicationController
   def index
     @user = User.find(params[:user_id])
     @posts = @user.posts
+    @comments = Comment.all
+    @users = User.all
```
Comment (ShadyShawkat) [inline app/controllers/posts_controller.rb:6]: - [ ] Kindly implement a solution for the N+1 problem. Currently, you're getting back all of the comments and users which isn't the solution. You can use [`includes`](https://apidock.com/rails/ActiveRecord/QueryMethods/includes) to fix the N+1 problem.

**Review by ShadyShawkat — CHANGES_REQUESTED** [review]
## Hi @ahmed025566, 

Good job so far! 👍🏻 
There are some issues that you still need to work on to go to the next project but you are almost there!

## To Highlight
- Correct use of GitFlow ✔️ 
- Descriptive commit messages ✔️ 
- Professional looking pull request ✔️ 

Image

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/controllers/posts_controller.rb (hunk 1 above)
Comment (Abenezer-Tilahun) [reply app/controllers/posts_controller.rb:6]: Fixed.

**Review by Abenezer-Tilahun — APPROVED** [review]
Hi @ahmed025566 @ClintonYounge 
 
Your project is complete  👍 💯   There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights 👏🏻

✔️ No linter errors.
✔️ You used correct Git-Flow.
✔️ Professional README  
✔️ Good Commit Messages
✔️ Good job for Implemented Suggested changes.

## Optional suggestions 

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

**Pull request merged**